diesel/build.rs

```diff
@@ -0,0 +1,11 @@
+fn main() {
+    #[cfg(feature = "sqlite")]
+    {
+        #[cfg(feature = "libsqlite3-sys/with-asan")]
+        println!("cargo:rustc-cfg=sqlite_supports_returning");
+        #[cfg(not(feature = "libsqlite3-sys/with-asan"))]
```

Comment:
The ASAN feature is causing linker errors for the build script. Unfortunately, it doesn't seem possible to compile code conditionally based on features of a dependency. I think this is a dead end unless someone has an idea how to work around this.

I see two alternative approaches that should work:

1. I could implement a proc macro attribute instead which would eliminate the build script and, as a consequence, the ASAN linker errors, too. But the proc macro would then depend on `libsqlite3-sys`. Also, it would require the introduction of a new crate because it doesn't really fit in `diesel_derives` nor in `diesel_migrations`.
2. We add an additional feature flag as [discussed here](https://github.com/diesel-rs/diesel/discussions/2684#discussioncomment-664045) along with the disadvantages it comes along.

@weiznich what do you think?

Reply:
The first solution would have the same problem as the build script based approach. You would need to build sqlite for the host target with different flags than for the target system. The asan feature passes the the corresponding flags to the compiler unconditionally for both targets. That will remain the same if its a proc macro instead. That in turn means that a feature flag based solution is the only possible solution, at least as long as we dont emulate the returning clause on older sqlitr versions somehow...

Reply:
> at least as long as we dont emulate the returning clause on older sqlitr versions somehow...

I was pondering the same. Old sqlite version could implicitly implement `explicit_returning()` from `examples/sqlite/all_about_inserts/src/lib.rs`. But I see two major downsides:

1. We're hiding from the user that we're opening a transaction and that we're hitting the db twice. Users can end up nesting transactions unknowingly. 
2. Users can choose `id` to be whatever. So we cannot rely it to be an incrementing integer. There could be tables that don't have an `id` column at all. This renders the `explicit_returning()` code too fragile for a general use case.

Old sqlite users have no other choice but to implement something a la `explicit_returning()` I'm afraid. 

I'll refactor the PR to use a feature flag.

Reply:
> We're hiding from the user that we're opening a transaction and that we're hitting the db twice. Users can end up nesting transactions unknowingly

I haven't tried that yet, but maybe a CTE could help there to avoid the transaction.

> Users can choose id to be whatever. So we cannot rely it to be an incrementing integer. There could be tables that don't have an id column at all. This renders the explicit_returning() code too fragile for a general use case.

We could use the "hidden" rowid column for that.

Both combined would result in the following query:

```sql
WITH 
INSERT ... AS insert
SELECT returning FROM table ORDER BY rowid DESC LIMIT count
```

Reply:
I've already checked that earlier. `rowid` is not guaranteed to exist :(

![image](https://user-images.githubusercontent.com/7702854/147757578-83a2e042-6bce-41dd-b1cf-23530640b7e2.png)

Reply:
I know. The important point for me is that it exists by default. Thant means people that are unaware about the details will not disable it. I would expect that only a minority disables this feature and that this minority is very aware of technical details. That means documenting this behavior may be enough. There are already a few other places in diesel that will not work if you use a too old db version, that wouldn't be that different in my opinion.

Reply:
I should probably add that I would like to experiment with this a bit, but that will likely need to wait till mid of January.

Reply:
Alright, thx for the heads up!